experimental/client-app/server/src/index.js

```diff
@@ -0,0 +1,162 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/* eslint-disable */
+import express from 'express';
+import jwt from 'jsonwebtoken';
+import axios from 'axios';
+import cors from 'cors';
+import querystring from 'querystring';
+import cookieParser from 'cookie-parser';
+import {
+  CLIENT_ID,
+  JWT_SECRET,
+  CLIENT_SECRET,
+  COOKIE_NAME,
+  REDIRECT_URI,
+  SITE_URL,
+} from './config.js';
+
+const port = 4000;
+const redirectURI = '/auth/google';
+const app = express();
+
+function getTokens({ code, clientId, clientSecret, redirectUri }) {
+  const url = 'https://oauth2.googleapis.com/token';
+  const values = {
+    code,
+    client_id: clientId,
+    client_secret: clientSecret,
+    redirect_uri: redirectUri,
+    grant_type: 'authorization_code',
+  };
+
+  const options = {
+    headers: {
+      'Content-Type': 'application/x-www-form-urlencoded',
+    },
+  };
+
+  return axios
+    .post(url, querystring.stringify(values), options)
+    .then(res => res.data)
+    .catch(err => {
+      console.error(`Failed to fetch auth tokens`);
+      throw new Error(err.message);
+    });
+}
+
+/**
+ * Applying cors
+ * Origin: applies access-control-allow-origin to client app
+ * Client: applies access-control-allow-credentials to true
+ */
+app.use(
+  cors({
+    origin: SITE_URL,
+    credentials: true,
+  })
+);
+
+app.use(cookieParser());
+
+/**
+ * Handle Auth Callback
+ */
```

Comment:
Can/should we link to documentation on how this works?

Maybe something like [this](https://developers.google.com/identity/protocols/oauth2/web-server) - though @engelke might know a better page.